scripts/docker/buildpack-deps/Dockerfile.ubuntu2004

```
@@ -37,7 +37,7 @@ RUN set -ex; \
 		cmake ninja-build \
 		libboost-filesystem-dev libboost-test-dev libboost-system-dev \
 		libboost-program-options-dev \
-		libcvc4-dev libz3-static-dev \
+		libcvc4-dev libz3-static-dev z3-static \
 		; \
 	apt-get install -qy python3-pip python3-sphinx; \
```

If another rebuild happen to be needed for any reason, could you also remove Sphinx from here? 
```suggestion
	apt-get install -qy python3-pip; \
```

Related to https://github.com/ethereum/solidity/pull/11662#discussion_r669859115. Our `docs.sh` script reinstalls Sphinx anyway. This installs an ancient version (1.8.5, latest is 4.1.0) which actually causes problems because `pip` does not by default update packages if they're already installed. I wanted to change it myself but my PR fixes the problem in a different way so it's not strictly required. Still, would be nice not to have this package here any more.

---

Just this one?

---

Yeah. `python3-pip` is still needed. It's just Sphinx that's superfluous.